File: sdk-ts/examples/capture_smoke.ts
// Smoke example for the TS SDK *capture mode*. The app makes the upstream
// call itself via captureFetch — the SDK ships the event to ingest-api with
// source = sdk-ts. The orchestrator runs this with a unique tag in the path
// and verifies the event landed in ClickHouse.
//
// Run: ECHOPROXY_API_KEY=sk_test_demo ECHOPROXY_TAG=abc \
//      node --experimental-strip-types examples/capture_smoke.ts

import { Client as IngestClient } from "../src/ingest.ts";
import { captureFetch } from "../src/capture.ts";

const tag = process.env.ECHOPROXY_TAG ?? "ts-default";
const target = process.env.ECHOPROXY_EXAMPLE_TARGET ?? "http://upstream-mock:9000";

const client = new IngestClient({ flushIntervalMs: 200 });
const fetch = captureFetch(client);

const url = `${target}/api/users/sdkbench-ts-capture-${tag}`;
const res = await fetch(url);
const body = await res.text();
console.log(`ts sdk (capture): ${url} -> ${res.status} (${body.length} bytes)`);

// Drain the SDK buffer before exit so the smoke event is shipped.
client.close();
await new Promise((r) => setTimeout(r, 500));
process.exit(res.status === 200 ? 0 : 1);

File: sdk-python/examples/capture_smoke.py
"""Smoke example for the Python SDK *capture mode*.

Uses httpx with the echoproxy event hooks: the app makes the upstream call
itself, the SDK ships the event to ingest-api. The orchestrator runs this with
a unique tag in the path and verifies the event landed in ClickHouse with
source = sdk-python.
"""

from __future__ import annotations

import os
import sys

import httpx

import echoproxy
from echoproxy import httpx_hook


def main() -> int:
    tag = os.environ.get("ECHOPROXY_TAG", "py-default")
    target = os.environ.get("ECHOPROXY_EXAMPLE_TARGET", "http://upstream-mock:9000")

    client = echoproxy.Client()
    http = httpx.Client(event_hooks=httpx_hook.hooks(client))

    url = f"{target}/api/users/sdkbench-py-capture-{tag}"
    r = http.get(url, timeout=5)
    print(f"py sdk (capture): {url} -> {r.status_code} ({len(r.content)} bytes)")

    client.close()
    return 0 if r.status_code == 200 else 1


if __name__ == "__main__":
    sys.exit(main())
